### PR TITLE
to 1.20, upgrade go mod

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

1. to golang `1.20`
2. upgrade dependency

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [x] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

- Fixes #4

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [ ] Passed all pipeline checking

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ make test
 
 ## Dependency management
 
-The K8s athenz SIA project uses [Go modules](https://golang.org/cmd/go/#hdr-Modules__module_versions__and_more) to manage dependencies on external packages. This requires a working Go environment with version 1.18 or greater installed.
+The K8s athenz SIA project uses [Go modules](https://golang.org/cmd/go/#hdr-Modules__module_versions__and_more) to manage dependencies on external packages. This requires a working Go environment with version 1.20 or greater installed.
```
Comment (mlajkim) [inline CONTRIBUTING.md:34]:
```suggestion
The K8s athenz SIA project uses [Go modules](https://golang.org/cmd/go/#hdr-Modules__module_versions__and_more) to manage dependencies on external packages. This requires a working Go environment with version 1.20 or later installed.
```

**Review by mlajkim — COMMENTED** [review]
lgtm! fix is optional.

**Review by mlajkim — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (WindzCUHK) [reply CONTRIBUTING.md:34]: @mlajkim resolve in new PR: https://github.com/AthenZ/k8s-athenz-sia/pull/9#issue-1641688535

**Pull request merged**